connect/src/com/telenor/connect/id/ConnectIdService.java

```diff
@@ -58,6 +58,14 @@ public String getAccessToken() {
         return retrieveTokens().getAccessToken();
     }
 
+    public Date getAccessTokenExpirationTime() {
+        if (retrieveTokens() == null) {
+            return null;
+        }
+
+        return retrieveTokens().getExpirationDate();
```

Comment:
If I'm not misunderstanding something the `getString(…)` from `getSharedPreferences` also uses a cache internally, and blocks (is synchronous), so there shouldn't be race conditions here, I think?